gcp/modules/locust/swarm.tf

```diff
@@ -0,0 +1,100 @@
+resource "null_resource" "locust_swarm_session" {
+  depends_on = ["module.locust"]
+
+  triggers = {
+    nonce = "${var.nonce}"
+  }
+
+  provisioner "local-exec" {
+    command = <<EOF
+      RETRIES=10
+      RETRY_COUNT=1
+      WORKERS_READY=0
+      while [ "$WORKERS_READY" -lt "${var.locust_workers}" ]; do
+        echo "[Try $RETRY_COUNT of $RETRIES] Waiting for all Locust workers to join the master..."
+        WORKERS_READY=$(kubectl -n locust logs deployment/locust-master --tail 1 | grep -oE "Currently \d+ clients" | grep -oE "\d+")
+        if [ "$WORKERS_READY" == "" ]; then
+          WORKERS_READY=0
+        fi
+        echo "Number of ready workers: $WORKERS_READY out of ${var.locust_workers}!"
+        RETRY_COUNT=$(($RETRY_COUNT+1))
+        if [ "$RETRY_COUNT" -eq "$RETRIES" ] ; then
+          echo "Retry limit reached, giving up!"
+          exit 1
+        fi
+        sleep 10
+      done
+
+      LOCUST_URL=http://127.0.0.1:8089
+      kubectl -n locust port-forward deployment/locust-master 8089:8089 </dev/null &>/dev/null &
+      sleep 5
+
+      echo
+      echo "Starting Locust swarm with ${var.locust_users} users and hatch rate of ${var.locust_hatch_rate}!"
+      curl -s -XPOST $LOCUST_URL/swarm -d"locust_count=${var.locust_users}&hatch_rate=${var.locust_hatch_rate}"
+
+      echo
+      echo "USERS  RPS     STATUS"
+      SWARM_DURATION=${var.locust_swarm_duration}
+      while [ "$SWARM_DURATION" != "0" ]; do
```

Comment:
This will loop forever if `SWARM_DURATION` is 0 or, as is more likely to happen accidentally, empty:

```
/project # foo="" ; echo $((foo - 1))
-1
/project # foo= ; echo $((foo - 1))
-1
```

Reply:
What is your proposition here? Extensively check if `${var.locust_swarm_duration}` empty before while loop?

Reply:
Maybe there's a more elegant way but this should work (btw it's probably wise to put quotes around the template variable):

```
SWARM_DURATION="${var.locust_swarm_duration}"
SWARM_DURATION=${SWARM_DURATION:-0}
```

Reply:
In your case it would never enter the loop and swarm would be immediately terminated. Replaced with `SWARM_DURATION=$((${var.locust_swarm_duration}+10))`.

Reply:
To me, terminating immediately seems like a reasonable result if `var.locust_swarm_duration` is undefined.

Your new version means that if `var.locust_swarm_duration = 1` it will actually run for 11 seconds? That seems surprising to me.

Reply:
10 seconds is a minimal amount of time for Locust to start hatching the swarm and making requests, so it can then produce stats. I can change this behavior for `swarm_duration` check before loop if you prefer. I don't think it makes much difference, since `swarm_duration` has its default value and it is very unlikely that we somehow override it with emptiness.